sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/batch/implementation/BatchAsyncBatcher.java

```diff
@@ -0,0 +1,224 @@
+        }
+
+        checkNotNull(operation, "expected non-null operation");
+        checkNotNull(operation.getContext(), "expected non-null operation context");
+
+        if (this.operations.size() == this.maxBatchOperationCount) {
+            logger.info("Batch is full - Max operation count {} reached.", this.maxBatchOperationCount);
+            return false;
+        }
+
+        int itemByteSize = operation.getApproximateSerializedLength();
+
+        if (!this.operations.isEmpty() && itemByteSize + this.currentSize > this.maxBatchByteSize) {
+            logger.info("Batch is full - Max byte size {} reached.", this.maxBatchByteSize);
+            return false;
+        }
+
+        this.currentSize += itemByteSize;
+
+        // Operation index is in the scope of the current batch
+        operation.setOperationIndex(this.operations.size());
+        operation.getContext().setCurrentBatcher(this);
+        this.operations.add(operation);
+
+        return true;
+    }
+
+    void dispatchBatch(BatchPartitionMetric partitionMetric) {
+
+        checkState(interlockIncrementCheck.tryAcquire(), "failed to acquire dispatch permit");
+
+        this.createBatchRequestAsync()
+            .exceptionally(exception -> {
+                logger.error("exception: {} {}", exception.getMessage(), exception.getStackTrace());
+
+                // Exceptions happening during request creation, fail the entire list
+                this.failBatchOperations(this.operations, exception);
+                throw new CompletionException(exception);
+            }).thenAcceptAsync(batchRequest -> {
+
+                // Serialization might leave some pending operations out of the batch
+                PartitionKeyRangeServerBatchRequest serverRequest = batchRequest.getBatchRequest();
+                final List<ItemBatchOperation<?>> batchPendingOperations = batchRequest.getBatchPendingOperations();
+
+                try {
+                    // Any overflow goes to a new batch
+                    for (ItemBatchOperation<?> batchOperation : batchPendingOperations) {
+                        this.retrier.apply(batchOperation).exceptionally((exception) -> {
+                            batchOperation.getContext().fail(this, exception);
+                            return null;
+                        });
+                    }
+                } catch (Exception exception) {
+                    // If retrier throws some exception, fail the pending operation but try out the serverRequest
+                    this.failBatchOperations(batchPendingOperations, exception);
+                }
+
+                try {
+                   this.executeServerRequest(serverRequest, partitionMetric);
+                } catch (Throwable throwable) {
+                    // Exceptions happening during execution, fail all the tasks part of the request (excluding overflow)
+                    this.failBatchOperations(serverRequest.getOperations(), throwable);
```

Comment:
The C# version has a retry on 429s https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Batch/BatchAsyncBatcher.cs#L187, is this covered somehow?